google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreImpl.java

```diff
@@ -400,6 +402,36 @@ protected Entity computeNext() {
     }
   }
 
+  @Override
+  public ReserveIdsResponse reserveIds(Key... keys) {
+    ReserveIdsRequest.Builder requestPb = ReserveIdsRequest.newBuilder();
+    for (Key key : keys) {
+      if (key instanceof Key) {
+        key = (Key) IncompleteKey.fromPb(key.toPb());
```

Comment:
Why do we need the `instanceof` check and Key -> pb -> IncompleteKey -> Key dance here?

Reply:
thanks for pointing out this, it's look like unused, I have already simplified this.